tests/ui/drop/drop_order.rs

```diff
@@ -55,11 +60,18 @@ impl DropOrderCollector {
     }
 
     fn if_let(&self) {
+        #[cfg(not(edition2024))]
```

Comment:
Question [REV6]: do we want cfg'd code, or do we actually want revision-gated error annotations here? i.e.

```
//[not_edition2024]~^ ERROR xxxx
// ... but no error for edition 2024 with the feature gate etc.
```

Because there will be a blessed stderr for *each* revision, so I don't think cfg'd out code makes sense here?

Reply:
I would prefer to go with cfg'd code.

The context of this test is that it is supposed to be compiled and the program have to run and generate a print-out to say which value is dropped in which order. We would like to let `rustc` to continue to produce the Edition 2021 drop order faithfully and reliably in the future. We would like to also be assured that Edition 2024 drop order is "documented" in this test.

Reply:
Also I am switching to `cfg(edition2021)` because it is much clearer.

Reply:
Ah right okay. That seems fine.